Subsurface/Node/Common/Peripherals/inc/SPIInterface.h

```diff
@@ -0,0 +1,63 @@
+#ifndef NODE_SPIINTERFACE_H
+#define NODE_SPIINTERFACE_H
+
+#include "stm32f3xx_hal.h"
+
+class SPIInterface
+{
+private:
+
+    SPI_HandleTypeDef *m_hspi;
+    uint8_t m_dataReceived[20];   // Added an arbitrary amount of buffer space
```

Comment:
Upon a quick google, it seems the max size of an i2c packet is typically 32 bytes, change to that just in case